explorer/src/Pos/Explorer/Web/Server.hs

```diff
@@ -633,13 +638,65 @@ epochSlotSearch epochIndex slotIndex = do
     getPageHHsOrThrow
         :: (DB.MonadBlockDB SscGodTossing m, MonadThrow m)
         => EpochIndex
+        -> Int
         -> m [HeaderHash]
-    getPageHHsOrThrow epoch = getEpochBlocks epoch >>=
-        maybeThrow (Internal errMsg)
+    getPageHHsOrThrow epoch page =
+        getEpochBlocks epoch page >>= maybeThrow (Internal errMsg)
       where
         errMsg :: Text
-        errMsg = sformat ("No blocks on epoch "%build%" found!") epoch
+        errMsg = sformat ("No blocks on epoch "%build%" page "%build%" found!") epoch page
+
+-- | Search the blocks by epoch and epoch page number.
+epochPageSearch
+    :: ExplorerMode ctx m
+    => EpochIndex
+    -> Int
+    -> m (Int, [CBlockEntry])
+epochPageSearch epochIndex page = do
+
+    -- We want to fetch as many pages as we have in this @Epoch@.
+    epochPagesNumber <- getEpochPages epochIndex >>= maybeThrow (Internal "No epoch pages.")
+
+    -- Get pages from the database
+    -- TODO: Fix this Int / Integer thing once we merge repositories
```

Comment:
The repositories of `cardano-sl` and `cardano-sl-explorer` have long been merged. Please open an issue, assign it to yourself and update this comment with the issue number. The reasoning is that currently you are the only one who has an idea of what the problem is about. For example, I don't even understand from this phrase where the `Int` / `Integer` problem arises, let alone how to fix it.

Reply:
The issue has been opened - https://issues.serokell.io/issue/CSE-249